src/python_testing/TC_pics_checker.py

```diff
@@ -0,0 +1,152 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+import math
+
+import chip.clusters as Clusters
+from basic_composition_support import BasicCompositionTests
+from global_attribute_ids import GlobalAttributeIds
+from matter_testing_support import (AttributePathLocation, ClusterPathLocation, CommandPathLocation, FeaturePathLocation,
+                                    MatterBaseTest, async_test_body, default_matter_test_main)
+from mobly import asserts
+from spec_parsing_support import build_xml_clusters
+
+
+def attribute_pics(pics_base: str, id: int) -> str:
+    return f'{pics_base}.S.A{id:04X}'
+
+
+def accepted_cmd_pics(pics_base: str, id: int) -> str:
+    return f'{pics_base}.S.C{id:02X}.Rsp'
+
+
+def generated_cmd_pics(pics_base: str, id: int) -> str:
+    return f'{pics_base}.S.C{id:02X}.Tx'
+
+
+def feature_pics(pics_base: str, bit: int) -> str:
+    return f'{pics_base}.S.F{bit:02X}'
+
+
+class TC_PICS_Checker(MatterBaseTest, BasicCompositionTests):
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+        await self.setup_class_helper(False)
+        self.xml_clusters, self.problems = build_xml_clusters()
+
+    def _check_and_record_errors(self, location, required, pics):
+        if required and not self.check_pics(pics):
+            self.record_error("PICS check", location=location, problem=f"Required PICS {pics} not found in pics list")
```

Comment:
"Required" here means that it is found on the device but not in the PICS list?

Reply:
Changed the wording here to be more verbose. But yes, the element is on the device, but the PICS is not in the list.